src/server/graphql/models/Project/projectMutation.js

```diff
@@ -52,25 +51,40 @@ export default {
         const [userId] = teamMemberId.split('::');
         newProject.userId = userId;
       }
-      await r.table('Project').get(id).update(newProject)
-        .do(() => {
-          return r.table('ProjectHistory')
-            .between([id, r.minval], [id, r.maxval], {index: 'projectIdUpdatedAt'})
-            .orderBy({index: 'projectIdUpdatedAt'})
-            .nth(-1)
-            .default({updatedAt: 0})
-            .do((lastDoc) => {
-              return r.branch(
-                lastDoc('updatedAt').gt(r.epochTime((now - DEBOUNCE_TIME) / 1000)),
-                r.table('ProjectHistory').get(lastDoc('id')).update(mergeDoc),
-                r.table('ProjectHistory').insert(lastDoc.merge(mergeDoc, {id: shortid.generate()}))
-              );
-            });
-        });
+      const projectUpdatePromise = r.table('Project').get(projectId).update(newProject);
+      const dbWork = [projectUpdatePromise];
```

Comment:
there are 2 optional DB hits in this mutation. Writing to the ProjectHistory and rebalancing the sort order. For each, I perform a test to see if we should do it, and if so, add it to the promise array.